scripts/components.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python3
+
+import argparse
+from copy import deepcopy
+import os
+import sys
+
+from schema import SchemaError
+import yaml
+
+from lib.schemata import (
+    component_schema, repo_url_schema, version_sha_schema, version_id_schema
+)
+
+
+def load_data(filepath, schema):
+    try:
+        with open(filepath) as f:
+            raw_data = yaml.safe_load(f)
+    except FileNotFoundError:
+        data = None
+    else:
+        data = schema.validate(raw_data)
+
+    return data
+
+
+def save_data(filepath, data, schema):
+        validated_data = schema.validate(data)
+        with open(filepath, "w") as f:
+            yaml.dump(validated_data, f, default_flow_style=False)
+
+
+def load_component(name, directory):
+    filename = "{name}.yml".format(name=name)
+    filepath = os.path.join(directory, filename)
+
+    return load_data(filepath, component_schema)
+
+
+def save_component(component, directory):
+    filename = "{name}.yml".format(name=component["name"])
+    filepath = os.path.join(directory, filename)
+
+    save_data(filepath, component, component_schema)
+
+
+def update_component(existing, name, repo_url, is_product):
+    if existing:
+        component = deepcopy(existing)
+    else:
+        component = {
+            "name": None,
+            "repo_url": None,
+            "is_product": None,
+            "releases": [],
+        }
+    new = {
+        k: v
+        for k, v in (
+            ("name", name),
+            ("repo_url", repo_url),
+            ("is_product", is_product)
+        )
+        if v is not None
+    }
+
+    component.update(**new)
+
+    return component_schema.validate(component)
+
+
+def update_releases(component, series_name, version, sha):
+    component = deepcopy(component)
+    release = {
+        "version": version,
+        "sha": sha,
+    }
+
+    for r in component["releases"]:
+        if r["series"] == series_name:
+            series_versions = r["versions"]
+            break
+    else:
+        series_versions = []
+        component["releases"].append(
+            {
+                "series": series_name,
+                "versions": series_versions,
+            }
+        )
+
+    series_versions.append(release)
+
+    return component_schema.validate(component)
+
+
+def parse_args(args):
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("--component-name", required=True)
+    parser.add_argument("--component-dir", default="./")
+
+    subparsers = parser.add_subparsers(dest="subparser")
+    subparsers.required = True
+
+    c_parser = subparsers.add_parser("component")
+
+    c_parser.add_argument("--repo-url", type=repo_url_schema.validate)
```

Comment:
It looks like `--repo-url` is required when subparser is `component`.

Reply:
It wasn't always required, I've split the functionality up to make that clear.